feature/google/src/main/java/com/revenuecat/purchases/google/purchaseOptionConversions.kt

```diff
@@ -5,5 +5,5 @@ import com.revenuecat.purchases.models.GooglePurchaseOption
 
 fun ProductDetails.SubscriptionOfferDetails.toPurchaseOption(): GooglePurchaseOption {
     val pricingPhases = pricingPhases.pricingPhaseList.map { it.toRevenueCatPricingPhase() }
-    return GooglePurchaseOption(pricingPhases, offerTags, offerToken)
+    return GooglePurchaseOption(offerId ?: basePlanId, pricingPhases, offerTags, offerToken)
```

Comment:
I added an `id` to `PurchaseOption` and I think what I did here makes sense. If there's an offerId in the `ProductDetails.SubscriptionOfferDetails`, that will be the `id` of the `PurchaseOption`, if not, use the `basePlanId`

Reply:
NABD. I think that makes sense but should we extract this to a different extension function for cleanliness? Something like `offerDetailsId`